physical/azure.go

```diff
@@ -59,12 +59,23 @@ func newAzureBackend(conf map[string]string, logger log.Logger) (Backend, error)
 	}
 
 	client, err := storage.NewBasicClient(accountName, accountKey)
-
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create Azure client: %v", err)
+		return nil, fmt.Errorf("failed to create Azure client: %v", err)
 	}
 
-	client.GetBlobService().CreateContainerIfNotExists(container, storage.ContainerAccessTypePrivate)
+	contObj := client.GetBlobService().GetContainerReference(container)
+	created, err := contObj.CreateIfNotExists()
+	if err != nil {
+		return nil, fmt.Errorf("failed to upsert container: %v", err)
+	}
+	if created {
+		err = contObj.SetPermissions(storage.ContainerPermissions{
+			AccessType: storage.ContainerAccessTypePrivate,
+		}, 0, "")
```

Comment:
https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations